tests/python/network_spammer.py

```diff
@@ -0,0 +1,236 @@
+
+    All prioritized transactions are expected to be included in the next ledger.
+    Only one out of all unprioritized transactions is expected to be included in the next ledger.
+
+    Return a result list and hash of all submitted transactions.
+    """
+    payment_amount = 1
+    payment_dest = prioritizers[0]  # all transactions are payments to the same address
+
+    # this round should be started only after all previous ledgers have been
+    # added, so we sleep until then
+    await asyncio.sleep(rnd * AVG_BLOCK_TIME)
+
+    # generate unprioritized payment transactions
+    # we submit them first because we want to test if prioritized transactions
+    # actually get priority over them
+    tx_metadata = {}
+    xdrs = []
+    for i, _ in enumerate(unprioritized_builders):
+        builder = unprioritized_builders[i]
+        builder.append_payment_op(payment_dest.public_address, str(payment_amount))
+        builder.sign(builder.keypair.seed().decode())
+
+        xdrs.append(builder.gen_xdr())
+
+        # cache tx hash for later review
+        tx_metadata[builder.hash_hex()] = {'round': rnd,
+                                           'prioritized': False}
+
+    # generate prioritized transactions
+    for i, _ in enumerate(prioritized_builders):
+        builder = prioritized_builders[i]
+        builder.append_payment_op(payment_dest.public_address, str(payment_amount))
+        builder.sign(builder.keypair.seed().decode())
+
+        # prioritize transaction by adding a prioritizer signature
+        builder.sign(prioritizers[i].secret_seed)
+
+        xdrs.append(builder.gen_xdr())
+
+        tx_metadata[builder.hash_hex()] = {'round': rnd,
+                                           'prioritized': True}
+
+    # remember submission time and send txs
+    submission_time = time.time()
+
+    # some transactions can fail with HTTP 500 Server Error or HTTP 504 Server
+    # Timeout, so don't raise an exception if this happens
+    tx_results = await send_txs(env.horizon_uri, xdrs, expected_statuses=[200, 500, 504])
+
+    # fetch builder sequence number.
+    #
+    # NOTE this is necessary because not all transactions have been processed
+    # when we reach this stage,specifically unprioritized ones.
+    # thus, it is unknown if the sequence has increased or not
+    all_builders = prioritized_builders + unprioritized_builders
+    sequences = await get_sequences(prioritized_builders[0].horizon.horizon_uri,
+                                    [b.address for b in all_builders])
+
+    # update builder sequence number
+    for i, _ in enumerate(all_builders):
+        b = all_builders[i]
+        b.clear()
+        b.sequence = str(sequences[i])
+
+    # create tx result object for each tx for reviewing later on
+    results = []
+    for (hsh, metadata), result in zip(tx_metadata.items(), tx_results):
+        try:
+            ledger = result['ledger']
+        except KeyError:  # probably 504
```

Comment:
I think that checking it will be safer than assuming its 504

Reply:
good idea